src/hierarchy/Reptile.ts

```diff
@@ -7,8 +7,34 @@ export default abstract class Reptile extends Animal implements ICanSwim{
         console.log(`${this.name} is swiming`);
       }
 
-    giveBirth(): void{
-
-    };
+      giveBirth(): [string, number, boolean, string][] {
+        if (this.isMale) {
```

Comment:
is that check not made for both reptiles and mammals? If only we had a common class for both types of animals where we could write the code only once...